File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        Arrays.sort(targets, (o1 ,o2) -> {
            return o1[1] - o2[1];
        });
        
        int prev = 0;
        for (int i = 0; i < targets.length; i++) {
            if (targets[i][0] >= prev) {
                answer++;
                prev = targets[i][1];
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(vector<int> &v1, vector<int> &v2) {
    return v1[1] < v2[1];
}
int solution(vector<vector<int>> targets) {
    int answer = 0;
    sort(targets.begin(), targets.end(), cmp);
    
    int point = targets[0][1];
    answer++;
    
    for (int i = 1; i < targets.size(); i++) {
        if (point <= targets[i][0]) {
            point = targets[i][1];
            answer++;
        }
    }
    return answer;
}